fixes for signedness compile errors with clang

lhoward opened the pull request:
NB: tested only on 64-bit platforms

lhoward commented:
We are using clang by default, I can of course add exceptions for gcc but it would be nice to be able to use clang.

vpodzime commented:
Hello, @lhoward! Thank you for your contribution! However, as you can see, the CI checks fail and it's because of the following compilation warning/error:

```
/builds/Northern.tech/Mender/mender-flash/libflash/optimized_writer.cpp: In member function 'mender::common::error::Error mender::io::OptimizedWriter::Copy(bool)':
/builds/Northern.tech/Mender/mender-flash/libflash/optimized_writer.cpp:49:32: error: comparison of integer expressions of different signedness: 'uint64_t' {aka 'long unsigned int'} and 'int64_t' {aka 'long int'} [-Werror=sign-compare]
   49 |    if (volumeSize_ && position < volumeSize_) {
      |                       ~~~~~~~~~^~~~~~~~~~~~~
/builds/Northern.tech/Mender/mender-flash/libflash/optimized_writer.cpp:60:55: error: comparison of integer expressions of different signedness: 'uint64_t' {aka 'long unsigned int'} and 'int64_t' {aka 'long int'} [-Werror=sign-compare]
   60 |   } else if (volumeSize_ && position + result.value() > volumeSize_) {
      |                             ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```
I get the same error when compiling locally (using GCC, on an x86_64 Linux system). Could you please address that? And while you are at it, could you please try to check your changes with a 32-bit compilation? It can be easily triggered by adding 
```
add_compile_options(-m32)
add_link_options(-m32)
```
to the `CMakeLists.txt` file (then you need to run `rm CMakeCache.txt && cmake .` or something similar).

lhoward commented:
Will do!

lhoward commented:
Checking in on this.

vpodzime commented:
> Checking in on this.

Thanks, @lhoward! If it turns out to be too difficult, please not spend too much time on this. We are hoping to be able to do something like what was suggested in #3 in the upcoming weeks/months.

lhoward commented:
I'm just waiting for review and integration, I believe I addressed your concerns. However if you are planning to rewrite in C then I will just maintain a local patch and close this issue.

vpodzime commented:
> I'm just waiting for review and integration, I believe I addressed your concerns. However if you are planning to rewrite in C then I will just maintain a local patch and close this issue.

Thanks. I'll see if I can make your changes pass CI and if that's reasonably easy, I'll merge it. Otherwise I'll close it and make sure the future C version is properly tested with Clang.

vpodzime commented:
> I'll see if I can make your changes pass CI and if that's reasonably easy, I'll merge it. Otherwise I'll close it

The CI is failing because of code formatting and commit message linting, both easy to fix for me. However, when I use clang and `-m32`, I still get compilation errors. Let's do simpler things in C and make sure they work great. Thanks for your contribution and patience @lhoward! I'll make sure to cover clang in the C version properly thanks to you.
